server/plugin.go

```diff
@@ -540,6 +543,16 @@ func (p *Plugin) GetRemoteID() string {
 }
 
 func (p *Plugin) updateMetrics() {
+	defer func() {
```

Comment:
This was run in an implicit goroutine already, so just tracking it like a proper worker.